### PR TITLE
e2e(FR-2504): add E2E tests for RBAC role management

### PR DESCRIPTION
**Pull request by ironAiken2**
Resolves #6530 ([FR-2504](https://lablup.atlassian.net/browse/FR-2504))

## Summary
- Add E2E tests for RBAC role CRUD lifecycle (create, edit, rename, deactivate, delete)
- Add E2E tests for RBAC role list view (table rendering, filtering, sorting, pagination)
- Add E2E tests for RBAC role detail drawer (permissions management, user assignments, edge cases)

## Test plan
- [ ] Run `npx playwright test e2e/rbac/` against a Backend.AI cluster with RBAC enabled
- [ ] Verify all 3 spec files pass: `rbac-role-crud`, `rbac-role-list`, `rbac-role-detail`

[FR-2504]: https://lablup.atlassian.net/browse/FR-2504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#6531** <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/6531?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/6531?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#6882** <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/6882?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>flow:merge-queue</kbd> - adds this PR to the back of the merge queue
- <kbd>flow:hotfix</kbd> - for urgent changes, fast-track this PR to the front of the merge queue

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/lablup?ref=merge-queue-instructions-in-stack-comment&prId=6531).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: e2e/rbac/rbac-role-list.spec.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
+// spec: e2e/.agent-output/test-plan-rbac-management.md
+// Scenarios: 1.1 – 1.4, 6.1, 6.4, 6.5 (Role list view, filtering, sorting, refresh)
+import { loginAsAdmin, navigateTo } from '../utils/test-util';
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-list.spec.ts:3]: Good point. Updated E2E_COVERAGE_REPORT.md to include the RBAC route with all 22 features listed.

File: e2e/rbac/rbac-role-detail.spec.ts [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+      // 13. Verify the Permission Type field becomes enabled
+      await expect(addModal.getByLabel('Permission Type')).toBeEnabled({
+        timeout: 5000,
+      });
+
+      // 14. Select "Project" from the Permission Type dropdown
+      // (When scope is Domain, the available entity types are Project and User, not Domain)
+      await page.waitForTimeout(500);
+      await addModal.locator('#entityType').click();
+      await page
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-detail.spec.ts]: Good point. Replaced with explicit wait for the dropdown to be enabled and options to be visible.

File: e2e/rbac/rbac-role-detail.spec.ts [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+        .first()
+        .click();
+
+      // 15. Select a permission (operation) from the Permission dropdown
+      await expect(addModal.locator('#operation')).toBeEnabled({
+        timeout: 5000,
+      });
+      await page.waitForTimeout(500);
+      await addModal.locator('#operation').click();
+      await page
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-detail.spec.ts]: Fixed. Replaced with explicit wait for the operation dropdown to be enabled and options to be visible.

File: e2e/rbac/rbac-role-detail.spec.ts [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+        .locator(
+          '.ant-select-dropdown:not(.ant-select-dropdown-hidden) .ant-select-item-option',
+        )
+        .first()
+        .click();
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-detail.spec.ts]: Fixed in the latest commit. Applied a similar pattern to the suggestion.

File: e2e/rbac/rbac-role-detail.spec.ts [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+        )
+        .filter({ hasText: 'Project' })
+        .first()
+        .click();
+      await expect(addModal.locator('#operation')).toBeEnabled({
+        timeout: 5000,
+      });
+      await page.waitForTimeout(500);
+      await addModal.locator('#operation').click();
+      await page
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-detail.spec.ts]: Fixed. Replaced with condition-based waits for both entity type and operation dropdowns.

File: e2e/rbac/rbac-role-crud.spec.ts [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+          .locator('.ant-message-notice-wrapper')
+          .filter({ hasText: /Role created successfully/i }),
+      ).toBeVisible({ timeout: 10000 });
+
+      // 14. Verify the new role appears in the table with a "Custom" source tag
+      const roleRow = page.getByRole('row').filter({ hasText: ROLE_NAME });
+      await expect(roleRow).toBeVisible({ timeout: 10000 });
+      await expect(
+        roleRow.locator('.ant-tag').filter({ hasText: 'Custom' }),
+      ).toBeVisible();
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply e2e/rbac/rbac-role-crud.spec.ts]: Good point. Fixed in the latest commit — now validates creation via the success notification only.

**Review by agatha197 — CHANGES_REQUESTED** [review]
When I tested it, I failed

```
RBAC Role Permissions Management
```

tests due to permission selection failing.

**Comment by ironAiken2** [comment]
> When I tested it, I failed
>
> ```
> RBAC Role Permissions Management
> ```
>
> tests due to permission selection failing.

​Just fixed :)

**Review by agatha197 — CHANGES_REQUESTED** [review]
```
 4 failed
    [chromium] › e2e/rbac/rbac-role-crud.spec.ts:56:9 › RBAC Role CRUD › Superadmin can create a new custom role with name and description @rbac @critical @functional 
    [chromium] › e2e/rbac/rbac-role-detail.spec.ts:158:9 › RBAC Role Detail Drawer › Drawer shows "Role Assignments" and "Permissions" tabs @rbac @critical @functional 
    [chromium] › e2e/rbac/rbac-role-detail.spec.ts:262:9 › RBAC Role Permissions Management › Superadmin can add a permission to a role @rbac @critical @functional 
    [chromium] › e2e/rbac/rbac-role-detail.spec.ts:559:9 › RBAC Role Assignments Management › Superadmin can assign a user to a role @rbac @critical @functional 
```

almost done! can you check again?

**Comment by ironAiken2** [comment]
@agatha197 Addressed the 4 failing tests via the stacked PR #6882 + test updates on this branch:

**Schema alignment (PR #6882, parent in the stack)**
- Updated `RoleAssignmentFilter.roleId` usages to the new `UUIDFilter` shape (`{ equals: ... }`).
- Refreshed `data/schema.graphql` to match the current backend schema.
- Removed a redundant `scopeRole: adminRole` alias so the drawer query resolves the role once.

**Test updates on this PR**
- `rbac-role-crud.spec.ts — Superadmin can create a new custom role`: fills the newly required `Scope Type / Target` fields, and validates creation via the success notification only (no paginated-row assumption).
- `rbac-role-detail.spec.ts — Drawer tabs`: updated to assert the three tabs (`Scopes`, `Permissions`, `Role Assignments`) that the drawer now exposes.
- `rbac-role-detail.spec.ts — Add permission / Assign user`: replaced `waitForTimeout` sequencing with explicit `toBeEnabled` + dropdown-option visibility waits to remove the flakiness you were hitting on permission selection.

Could you re-run the failing specs once more? Thanks!

**Review by agatha197 — APPROVED** [review]
LGTM
